code/game/objects/items/weapons/swords_axes_etc.dm

```diff
@@ -83,6 +91,19 @@
 		add_blood(blood_color)
 	return
 
+/obj/item/weapon/melee/telebaton/proc/stun(mob/living/carbon/human/target, mob/living/user)
+	if(target.check_shields(src, 0, "[user]'s [name]"))
```

Comment:
for this, can you use named parameters, since it's unclear what "0" and "[user]'s [name]" is supposed to be. By this I mean where you do check_shields(src, damage = 0, attack-text = "[user]'s [name]")